src/providers/music.rs

```diff
@@ -184,9 +184,12 @@ impl MediaPlayerRenderer {
 
         #[cfg(not(target_os = "windows"))]
         {
-            let length = metadata
-                .length()
-                .map_err(|_| anyhow!("Couldn't get length!"))? as f64;
+            let length_result = metadata.length();
+            let length = match length_result {
```

Comment:
This could be [`unwrap_or_default`](https://doc.rust-lang.org/std/result/enum.Result.html#method.unwrap_or_default) or [`unwrap_or`](https://doc.rust-lang.org/std/result/enum.Result.html#method.unwrap_or) instead.

Reply:
done